client/my-sites/email/email-providers-stacked-comparison/provider-cards/utils.tsx

```diff
@@ -0,0 +1,157 @@
+			'{{intervalLength/}} is already translated and it is either annually or monthly',
+	} );
+};
+
+export const addToCartAndCheckout = (
+	shoppingCartManager: ShoppingCartManagerActions,
+	cartItem: RequestCartProduct | IncompleteRequestCartProduct,
+	productList: Record< string, { product_id: number } >,
+	setAddingToCart: ( addingToCart: boolean ) => void,
+	selectedSite: string
+): void => {
+	setAddingToCart( true );
+
+	shoppingCartManager
+		.addProductsToCart( [ fillInSingleCartItemAttributes( cartItem, productList ) ] )
+		.then( ( response: ResponseCart ) => {
+			setAddingToCart( false );
+			const errors = response?.messages?.errors;
+			if ( errors && errors.length ) {
+				// Stay on the page to show the relevant error(s)
+				return;
+			}
+
+			page( `/checkout/${ selectedSite }` );
+		} )
+		.then( () => {
+			setAddingToCart( false );
+		} );
+};
+
+export const recordTracksEventAddToCartClick = (
+	comparisonContext: string,
+	validatedMailboxUuids: string[],
+	mailboxesAreValid: boolean,
+	provider: string,
+	source: string,
+	userCanAddEmail: boolean,
+	userCannotAddEmailReason: any
+) => {
+	recordTracksEvent( 'calypso_email_providers_add_click', {
+		context: comparisonContext,
+		mailbox_count: validatedMailboxUuids.length,
+		mailboxes_valid: mailboxesAreValid ? 1 : 0,
+		provider: provider,
+		source,
+		user_can_add_email: userCanAddEmail,
+		user_cannot_add_email_code: userCannotAddEmailReason ? userCannotAddEmailReason.code : '',
+	} );
+};
+
+type PriceWithIntervalProps = {
+	className: string;
+	cost: number;
+	currencyCode: string;
+	hasDiscount: boolean;
+	sale?: number | null;
+	intervalLength: IntervalLength;
+};
+
+export const PriceWithInterval: FunctionComponent< PriceWithIntervalProps > = ( props ) => {
+	const { className, cost, currencyCode, hasDiscount, sale, intervalLength } = props;
+
+	const costClassName = `${ className }__${ hasDiscount ? 'discounted-price' : 'keep-main-price' }`;
+	const intervalLengthClassName = `${ className }__interval`;
+	const mailboxClassName = `${ className }__mailbox`;
+	const saleClassName = `${ className }__sale`;
+
+	const showSale = sale && sale !== 0;
+
+	const lineThrough = {
+		textDecoration: hasDiscount || showSale ? 'line-through' : 'solid ',
+	};
+
+	const priceSpan = (
+		<span style={ lineThrough } className={ costClassName }>
+			{ formatCurrency( cost ?? 0, currencyCode ) }
+		</span>
+	);
+	const saleSpan = (
+		<span className={ saleClassName }>{ formatCurrency( sale ?? 0, currencyCode ) }</span>
+	);
+	const intervalLengthSpan =
+		intervalLength === 'annually' ? (
+			<span className={ intervalLengthClassName }>/{ translate( 'annually' ) }</span>
+		) : (
+			<span className={ intervalLengthClassName }>/{ translate( 'monthly' ) }</span>
+		);
+
+	const mailboxSpan = <span className={ mailboxClassName }> /{ translate( 'mailbox' ) } </span>;
+
+	return (
+		<>
+			{ priceSpan } { showSale && saleSpan } { mailboxSpan } { intervalLengthSpan }
```

Comment:
Since we didn't localize this whole block of text, I wonder if this is not going to be a problem for RTL languages...

Reply:
Hi @dlind1 👋. Do you think it's fine to break the price tag with multiple components/strings that are localized?

![image](https://user-images.githubusercontent.com/594356/145262753-ebeea19f-ab5f-4f95-bf6b-7b17212c8dad.png)

Or should we localize it [like this](https://github.com/Automattic/wp-calypso/blob/1300afbd1d71d27a2c564c6450781b5ef615d5e7/client/my-sites/email/email-providers-comparison/index.jsx#L384)?

Reply:
In this I'd recommend having a single string for the `price mailbox / interval` and another one for `price sale mailbox / interval` (like in your example) for a total of four strings, since otherwise it'd be almost impossible to produce naturally sounding translations. If you'd check [the translations](https://translate.wordpress.com/projects/wpcom/-all-translated/702448/) for your example you can see that there is some variation between the translations which would be impossible to have with concatenated strings.